### PR TITLE
fix: upload new linux file when it's outdated after autoupgrade of linux node

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind fix

**What this PR does / why we need it**:
upload new linux file when it's outdated after autoupgrade of linux node.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: e2e/windows/e2e-helper.sh [hunk 1]
```diff
@@ -85,6 +85,26 @@ upload_linux_file_to_storage_account() {
     fi
 }
 
+check_linux_file_outdated() {
```
Comment (ShiqianTao) [inline e2e/windows/e2e-helper.sh:88]: It's not just a `check` function; it seems more like an `update` function, right?

File: e2e/windows/e2e-helper.sh [hunk 2]
```diff
@@ -85,6 +85,26 @@ upload_linux_file_to_storage_account() {
...
+    
+    linuxFileURL="https://${AZURE_E2E_STORAGE_ACCOUNT_NAME}.blob.core.windows.net/${WINDOWS_E2E_STORAGE_CONTAINER}/${MC_VMSS_NAME}-linux-file.zip"
+
+    compare_message=$(az vmss run-command invoke --command-id RunShellScript \
+        --resource-group $E2E_MC_RESOURCE_GROUP_NAME \
+        --name $MC_VMSS_NAME \
+        --instance-id $VMSS_INSTANCE_ID \
+        --scripts "cat /etc/kubernetes/azure.json > /home/fields.json; cat /etc/kubernetes/certs/apiserver.crt | base64 -w 0 > /home/apiserver.crt; cat /etc/kubernetes/certs/ca.crt | base64 -w 0 > /home/ca.crt; cat /etc/kubernetes/certs/client.key | base64 -w 0 > /home/client.key; cat /var/lib/kubelet/bootstrap-kubeconfig > /home/bootstrap-kubeconfig; cd /home; zip new.zip fields.json apiserver.crt ca.crt client.key bootstrap-kubeconfig; wget https://aka.ms/downloadazcopy-v10-linux; tar -xvf downloadazcopy-v10-linux; cd ./azcopy_*; export AZCOPY_AUTO_LOGIN_TYPE=\"MSI\"; export AZCOPY_MSI_RESOURCE_STRING=\"${AZURE_MSI_RESOURCE_STRING}\"; ./azcopy copy $linuxFileURL /home/old.zip" | jq -r ".value[0].message")
+    
+    if [[ "$compare_message" == *"differ"* ]]; then
```
Comment (AbelHu) [inline e2e/windows/e2e-helper.sh:102]: Is `differ` returned by azcopy? Could you add a comment here?

Comment (junjiezhang1997) [reply e2e/windows/e2e-helper.sh:114]: I have updated this part of code and added the comments.

File: e2e/windows/e2e-helper.sh (hunk 1 above)
Comment (junjiezhang1997) [reply e2e/windows/e2e-helper.sh:88]: Thanks for the advice. I choose option 2 and update the code.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Pull request merged**